src/store/userSlice.test.ts

```diff
@@ -0,0 +1,30 @@
+import userReducer, { initialState, authUser } from './userSlice';
+
+describe('userSlice', () => {
+  describe('reducer', () => {
+    it('should handle authUser action', () => {
+      const username = 'JohnDoe';
+
+      const nextState = userReducer(initialState, authUser({ username }));
+
+      expect(nextState.userInfo).toEqual({ username });
```

Comment:
I think you will find that `expect(initialState.userInfo).toEqual({ username });` also passes, which basically means your input data is tainted by the reducer

Reply:
Never mind - previously your were using Immer behind the scenes, because that is what Redux Toolkit does, so the expectation I mention would have probably not passed.